metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/MetricCache.java

```diff
@@ -0,0 +1,141 @@
+package org.hypertrace.alert.engine.metric.anomaly.detector.evaluator;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.lang3.tuple.Pair;
+import org.hypertrace.alert.engine.eventcondition.config.service.v1.MetricSelection;
+import org.hypertrace.core.query.service.api.ResultSetChunk;
+import org.hypertrace.core.query.service.api.Row;
+import org.hypertrace.core.query.service.api.Value;
+import org.hypertrace.core.query.service.api.ValueType;
+
+class MetricCache {
+
+  private static final int CACHE_EXPIRY_MINUTES = 5;
+
+  private final Cache<MetricSelection, MetricTimeSeries> metricCache;
```

Comment:
Just curious Q: would be `MetricSelection` hash be costly?

Reply:
the hashcode is cached for the object, so it shouldn't cause a problem?